packages/clarity-js/src/layout/region.ts

```diff
@@ -42,23 +44,39 @@ export function track(id: number): void {
 }
 
 export function compute(): void {
+    // Process any regions that couldn't be processed earlier
+    let q = [];
+    for (let r of queue) {
+        let id = dom.getId(r.node);
+        if (id) {
+            r.data.id = id;
+            data.push(r.data);
+        } else { q.push(r); }
+    }
+    queue = q;
+
     // Schedule encode only when we have at least one valid data entry
     if (data.length > 0) { encode(Event.Region); }
 }
 
 function handler(entries: IntersectionObserverEntry[]): void {
     for (let entry of entries) {
         let target = entry.target;
-        let id = target ? dom.getId(target) : null;
-        if (id && regionMap.has(target)) {
+        let rect = entry.boundingClientRect;
+
+        // Only capture regions that are not hidden on the page
+        if (regionMap.has(target) && rect.width > 0 && rect.height > 0) {
```

Comment:
What if the region is off document? Off screen? Looked around to see how this was done and saw lot of methods, many of which are much more complicated than this. Some links:

[https://stackoverflow.com/questions/66070706/how-to-determine-if-html-element-is-generally-visible-javascript](https://stackoverflow.com/questions/66070706/how-to-determine-if-html-element-is-generally-visible-javascript) (lot of complicated logic)
[https://www.javascripttutorial.net/dom/css/check-if-an-element-is-visible-in-the-viewport/](https://www.javascripttutorial.net/dom/css/check-if-an-element-is-visible-in-the-viewport/) (use inner wdith/heigh, etc.)
[https://www.geeksforgeeks.org/how-to-check-if-an-element-is-visible-in-dom/](https://www.geeksforgeeks.org/how-to-check-if-an-element-is-visible-in-dom/) (sort of like your but more)

Why does interaction observer even return hidden elements? Seems odd. Also would any of the intersection observe entry properties be useful for this check (they already exist, no bounding box computation). 

Etc.

Reply:
We are using entry's bounding rectangle for the check - not recomputing. The goal here is to eliminate low hanging fruit - for instance, elements that are on the page but are display:none. Edge cases are both a significant hit to performance (computedStyles) and also not guaranteed. Plus we also don't need to cover all edge cases since these rules are for user defined regions and can be modified to tackle edge cases.